src/data_chain/chain.rs

```diff
@@ -185,3 +171,309 @@ impl Chain {
         }
     }
 }
+
+impl Debug for Chain {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        writeln!(formatter, "Chain {{")?;
+        writeln!(formatter, "    group_size: {},", self.group_size)?;
+        writeln!(formatter, "    path: {:?},", self.path)?;
+        writeln!(formatter, "    blocks:")?;
+        for block in &self.blocks {
+            writeln!(
+                formatter,
+                "        {:?} signed by {:?},",
+                block.payload(),
+                block.proofs()
+            )?;
+        }
+        writeln!(formatter, "    valid_nodes:")?;
+        for block in &self.valid_nodes {
+            writeln!(
+                formatter,
+                "        {:?},",
+                block.payload(),
+            )?;
+        }
+        writeln!(formatter, "    data:")?;
+        for block in &self.data {
+            writeln!(
+                formatter,
+                "        {:?} signed by {:?},",
+                block.payload(),
+                block.proofs()
+            )?;
+        }
+        writeln!(formatter, "    pending_blocks:")?;
+        for block in &self.pending_blocks {
+            writeln!(
+                formatter,
+                "        {:?} signed by {:?},",
+                block.payload(),
+                block.proofs()
+            )?;
+        }
+        writeln!(formatter, "}}")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::super::tests;
+    use data_chain::{NodeState, Proof, State};
+    use full_info::FullInfo;
+    use maidsafe_utilities::SeededRng;
+    use rand::Rng;
+    use routing_table::Prefix;
+    use rust_sodium;
+
+    #[ignore]
+    #[test]
+    fn pending_blocks() {
+        // Start with the chain having Live(A).  Create blocks for Offline(A), Live(B), and
+        // Relocated(B), Live(C).  Add the four of them in reverse order and assert after each is
+        // added that they're held in the `pending_blocks` until the final one is added and they can
+        // all be added to the chain in the order defined above.
```

Comment:
I think this scenario is assuming the chain contains GROUP_SIZE elder blocks already,
otherwise, a `Live(C)` is valid after a `Live(A)` ?
However, the testing code seems only having one live block in the chain?
Will it be better to make it explicit in the comment ?

Reply:
I think we can update that once we have agreement on how the chain will be getting validated (if at all) inside the constructor and how pruning will work.  Since the blocks are relating to a non-empty prefix, we're out of the startup phase, and hence two `Live` blocks shouldn't be allowed next to each other.